shell/platform/darwin/ios/framework/Source/FlutterDartProject.mm

```diff
@@ -89,14 +101,19 @@
     // In case the application bundle is still not specified, look for the App.framework in the
     // Frameworks directory.
     if (settings.application_library_path.size() == 0) {
-      NSString* applicationFrameworkPath = [mainBundle pathForResource:@"Frameworks/App.framework"
-                                                                ofType:@""];
-      if (applicationFrameworkPath.length > 0) {
-        NSString* executablePath =
-            [NSBundle bundleWithPath:applicationFrameworkPath].executablePath;
-        if (executablePath.length > 0) {
-          settings.application_library_path.push_back(executablePath.UTF8String);
-        }
+      NSString* libraryPath = FindFrameworkApplicationLibraryPath(mainBundle);
+      if (libraryPath.length > 0) {
+        settings.application_library_path.push_back(libraryPath.UTF8String);
+      }
+    }
+  }
+
+  // Check other bundles if we still haven't found anything.  This is useful for XCTest targets.
```

Comment:
Can you describe this issue a bit? Where does the framework end up and what does the file structure look like in a test case?

Reply:
Sure, it's kind of goofy, here is the file structure of the app that will run:
```
ScenariosUITests-Runner.app
ScenariosUITests-Runner.app/_CodeSignature
ScenariosUITests-Runner.app/_CodeSignature/CodeResources
ScenariosUITests-Runner.app/ScenariosUITests-Runner
ScenariosUITests-Runner.app/PlugIns
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest/_CodeSignature
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest/_CodeSignature/CodeResources
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest/ScenariosUITests
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest/Frameworks
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest/Frameworks/Flutter.framework
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest/Frameworks/Flutter.framework/_CodeSignature
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest/Frameworks/Flutter.framework/_CodeSignature/CodeResources
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest/Frameworks/Flutter.framework/icudtl.dat
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest/Frameworks/Flutter.framework/Flutter
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest/Frameworks/Flutter.framework/Info.plist
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest/Frameworks/App.framework
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest/Frameworks/App.framework/_CodeSignature
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest/Frameworks/App.framework/_CodeSignature/CodeResources
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest/Frameworks/App.framework/App
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest/Frameworks/App.framework/Info.plist
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest/Info.plist
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest.dSYM
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest.dSYM/Contents
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest.dSYM/Contents/Resources
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest.dSYM/Contents/Resources/DWARF
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest.dSYM/Contents/Resources/DWARF/ScenariosUITests
ScenariosUITests-Runner.app/PlugIns/ScenariosUITests.xctest.dSYM/Contents/Info.plist
ScenariosUITests-Runner.app/Frameworks
ScenariosUITests-Runner.app/Frameworks/XCTAutomationSupport.framework
ScenariosUITests-Runner.app/Frameworks/XCTAutomationSupport.framework/_CodeSignature
ScenariosUITests-Runner.app/Frameworks/XCTAutomationSupport.framework/_CodeSignature/CodeResources
ScenariosUITests-Runner.app/Frameworks/XCTAutomationSupport.framework/XPCServices
ScenariosUITests-Runner.app/Frameworks/XCTAutomationSupport.framework/XPCServices/LogArchiveCollector.xpc
ScenariosUITests-Runner.app/Frameworks/XCTAutomationSupport.framework/XPCServices/LogArchiveCollector.xpc/_CodeSignature
ScenariosUITests-Runner.app/Frameworks/XCTAutomationSupport.framework/XPCServices/LogArchiveCollector.xpc/_CodeSignature/CodeResources
ScenariosUITests-Runner.app/Frameworks/XCTAutomationSupport.framework/XPCServices/LogArchiveCollector.xpc/LogArchiveCollector
ScenariosUITests-Runner.app/Frameworks/XCTAutomationSupport.framework/XPCServices/LogArchiveCollector.xpc/Info.plist
ScenariosUITests-Runner.app/Frameworks/XCTAutomationSupport.framework/XPCServices/LogArchiveCollector.xpc/version.plist
ScenariosUITests-Runner.app/Frameworks/XCTAutomationSupport.framework/Info.plist
ScenariosUITests-Runner.app/Frameworks/XCTAutomationSupport.framework/XCTAutomationSupport
ScenariosUITests-Runner.app/Frameworks/XCTAutomationSupport.framework/version.plist
ScenariosUITests-Runner.app/Frameworks/XCTest.framework
ScenariosUITests-Runner.app/Frameworks/XCTest.framework/_CodeSignature
ScenariosUITests-Runner.app/Frameworks/XCTest.framework/_CodeSignature/CodeResources
ScenariosUITests-Runner.app/Frameworks/XCTest.framework/en.lproj
ScenariosUITests-Runner.app/Frameworks/XCTest.framework/en.lproj/InfoPlist.strings
ScenariosUITests-Runner.app/Frameworks/XCTest.framework/XCTest
ScenariosUITests-Runner.app/Frameworks/XCTest.framework/Info.plist
ScenariosUITests-Runner.app/Frameworks/XCTest.framework/version.plist
ScenariosUITests-Runner.app/embedded.mobileprovision
ScenariosUITests-Runner.app/Info.plist
ScenariosUITests-Runner.app/PkgInfo
```

`ScenariosUITests-Runner.app` is the application that gets run, but `Runner.app/PlugIns/ScenariosUITests.xctest` is the target you are building.  So, the code before this change tries to find the compiled app in the main bundle, ie `ScenariosUITests-Runner.app/Frameworks/App.framework`, but there is no way to actually put the framework in there.  So, we are making the code look through it's bundles so it find `Runner.app/PlugIns/ScenariosUITests.xctest/Frameworks/App.framework`

Reply:
I see. In this case, we should let the test itself ask to push in a custom bundle. From the production side, we should offer APIs to look at different bundles but the production code shouldn't go find them itself. i.e., doesn't FlutterViewController already have a initWithEngine: nibName: bundle: API? If the test knows what its own project structure should be, it should go tell the engine what to load.

Reply:
Yep, there is a way to specify the bundle directly.  I can do that if you want.  I don't really see a downside to making our search for the code a bit more robust when it only gets executed if all else fails and the client wasn't already specific.

Reply:
This would just iterate through their project bundles looking for one named App.framework right? It seems a bit too assumptive vs something explicit. If the users are at a point where their frameworks are sitting in random places in their .app, they probably should know exactly what they're doing.  

@jmagman might have opinions too.

Reply:
Started being explicit.